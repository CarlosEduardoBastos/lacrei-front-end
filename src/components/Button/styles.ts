import { NavLink } from 'react-router-dom';
import styled, { css, DefaultTheme } from 'styled-components';

export const Container = styled(NavLink)`
  ${({ theme, background }:{theme: DefaultTheme, background: Number}) => css`
    font-size: 1.125rem;
    width: 192px;
    height: 48px;
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${background ? theme.colors.primaryColor : 'transparent'};
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    color: ${background ? theme.colors.white : theme.colors.primaryColor};
    border-radius: 8px;
    border: ${background ? 'none' : '2px solid #018762'};
    transition: all 0.2s ease 0s;
    font-weight: 700;
    margin: 10px 0;

    &:hover {
      opacity: 0.7;
    }
  `}
`;
