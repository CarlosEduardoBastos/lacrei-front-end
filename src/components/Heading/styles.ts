import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xsmall};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xxlarge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media (max-width: 1039px) {
    font-size: ${theme.fonts.sizes.large};
  }
`;

export const Container = styled.h1<HeadingProps>`
  ${({ theme, size }) => css`
    color: ${theme.colors.secondaryColor};
    font-weight: bold;
    ${size ? titleSize[size](theme) : null}
  `}
`;
