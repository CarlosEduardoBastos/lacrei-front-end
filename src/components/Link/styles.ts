/* eslint-disable no-unused-vars */
import { NavLink } from 'react-router-dom';
import styled, { css } from 'styled-components';

const mediaQuery = () => css`
  @media (max-width: 768px) {
    font-size: 14px;
    padding: 0 0.625rem;
  }
`;

export const Container = styled(NavLink)`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryColor};
    font-size: ${theme.fonts.sizes.small};
    font-weight: 700;
    transition: all 0.2s ease 0s;
    padding: 0 1.25rem;
    line-height: 22px;

    &:hover {
      color: ${theme.colors.secondaryColorHover};
    }

    &.active {
      color: ${theme.colors.primaryColor};
    }
    ${mediaQuery()}
  `}
`;
