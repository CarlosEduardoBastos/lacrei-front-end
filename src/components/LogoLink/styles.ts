import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';

export const Container = styled(Link)`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.large};
    color: ${theme.colors.primaryColor};
    font-weight: 700;
    transition: all 0.2s ease 0s;
    line-height: 44px;

    &:hover {
      color: ${theme.colors.primaryColorHover};
    }
  `}
`;
