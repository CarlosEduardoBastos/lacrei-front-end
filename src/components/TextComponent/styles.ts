import styled, { css, DefaultTheme } from 'styled-components';

const mediaFont = (theme: DefaultTheme) => css`
  @media (max-width: 1039px) {
    font-size: ${theme.fonts.sizes.small};
    margin-top: 1rem;
  }
`;

export const Container = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.colorText};
    font-size: ${theme.fonts.sizes.medium};
    margin-top: 2rem;
    ${mediaFont(theme)}
  `}
`;
