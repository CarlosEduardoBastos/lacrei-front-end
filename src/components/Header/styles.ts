import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const mediaQuery = () => css`
  @media (max-width: 520px) {
    justify-content: center;
  }
`;

export const Container = styled.header`
  ${({ theme }) => css`
    width: 100%;
    background: ${theme.colors.colorHeader};
    padding: 0.5rem 4rem;

    ${SectionContainer} {
      padding: 0;
    }
  `}
`;

export const MenuContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  ${mediaQuery()}
`;

export const NavLink = styled.ul`
  display: flex;
  align-items: center;
  justify-content: center;
`;
