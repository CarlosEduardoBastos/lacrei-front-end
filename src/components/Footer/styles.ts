
import styled, { css } from 'styled-components';
import { Container as Link } from '../Link/styles';

export const Container = styled.footer`
  
`;

export const NavLink = styled.div`
  ${({ theme }) => css`
    ${Link} {
      opacity: 0.7;
      &.active {
        color: ${theme.colors.secondaryColor};
        opacity: 1;
      }
    }
  `}
`;

export const NavIcon = styled.div`
  margin: 24px 0;
`;

export const Icon = styled.a`
  padding: 0 1.25rem;
  
`;

export const Text = styled.p`
  ${({ theme }) =>
    css`
      font-size: 0.75rem;
      color: ${theme.colors.colorText};
    `}
`;

export const Content = styled.div`
  border-top: 1px solid #b0e0d3;
  padding-top: 2rem;
  
`;
