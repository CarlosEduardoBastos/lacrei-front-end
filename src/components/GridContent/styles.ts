import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

const mediaQuery = () => css`
  @media (max-width: 1088px) {
    flex-direction: column;
    align-items: center;
  }
`;

export const Container = styled.div`
  padding-top: 4rem;
`;

export const Grid = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  ${mediaQuery()};
`;
export const ContentText = styled.div`
  ${({ border }:{border: boolean}) => css`
    max-width: 453px;
    ${TextComponent} {
      ${() =>
        border
          ? css`
              border-left: 5px solid rgb(1, 135, 98);
              padding: 0 17px;
            `
          : ''}
    }
  `}
`;

export const ContentImage = styled.div`
  max-width: 540px;
`;
export const Image = styled.img`
  width: 100%;
`;
export const ContentButton = styled.div`
  max-width: 453px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 48px;
  flex-wrap: wrap;
  ${mediaQuery()}
`;
