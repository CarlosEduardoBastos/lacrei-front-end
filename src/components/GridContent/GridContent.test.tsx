import { screen } from '@testing-library/react';
import { GridContent } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<GridContent />', () => {
  it('should render a text a title and an image', () => {
    renderTheme(<GridContent title="title" text="text" src="image.img" />);
    expect(screen.getByRole('heading', { name: 'title' })).toBeInTheDocument();
    expect(screen.getByText('text')).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('should have two buttons on screen if haveButton is true', () => {
    renderTheme(<GridContent title="title" text="text" src="image.img" haveButton={true} />);
    expect(screen.getAllByRole('link')).toHaveLength(2);
  });

  it('should have zero buttons on screen if haveButton is false', () => {
    renderTheme(<GridContent title="title" text="text" src="image.img" haveButton={false} />);
    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<GridContent title="title" text="text" src="image.img" />);
    expect(container).toMatchSnapshot();
  });
});
