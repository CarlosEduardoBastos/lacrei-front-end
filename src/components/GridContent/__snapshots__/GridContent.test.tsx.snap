// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> should match snapshot 1`] = `
.c7 {
  font-size: 1.125rem;
  width: 192px;
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #018762;
  box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
  color: #FFFFFF;
  border-radius: 8px;
  border: none;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  font-weight: 700;
  margin: 10px 0;
}

.c7:hover {
  opacity: 0.7;
}

.c8 {
  font-size: 1.125rem;
  width: 192px;
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: transparent;
  box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
  color: #018762;
  border-radius: 8px;
  border: 2px solid #018762;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  font-weight: 700;
  margin: 10px 0;
}

.c8:hover {
  opacity: 0.7;
}

.c4 {
  color: #1f1f1f;
  font-weight: bold;
  font-size: 3rem;
}

.c1 {
  max-width: 85.375rem;
  margin: 0 auto;
  padding: 0 4rem;
}

.c5 {
  color: #515151;
  font-size: 1.5rem;
  margin-top: 2rem;
}

.c0 {
  padding-top: 4rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  max-width: 453px;
}

.c9 {
  max-width: 540px;
}

.c10 {
  width: 100%;
}

.c6 {
  max-width: 453px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 48px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (max-width:1039px) {
  .c4 {
    font-size: 2rem;
  }
}

@media (max-width:1039px) {
  .c5 {
    font-size: 1rem;
    margin-top: 1rem;
  }
}

@media (max-width:1088px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:1088px) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4"
          >
            title
          </h2>
          <p
            class="c5"
          >
            text
          </p>
          <div
            class="c6"
          >
            <a
              background="1"
              class="c7"
              href="/user-person"
            >
              Pessoa Usu√°ria
            </a>
            <a
              background="0"
              class="c8"
              href="/profissional"
            >
              Profissional
            </a>
          </div>
        </div>
        <div
          class="c9"
        >
          <img
            class="c10"
            src="image.img"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
